### ISSUE TITLE
gdasgldas task fails with Restart Tile Space Mismatch

### ISSUE THREAD

**Issue by BrettHoover-NOAA** [NOAA-EMC/global-workflow#622]
**Expected behavior**
gdasgldas task should complete successfully and global-workflow should continue to cycle fv3gdas

**Current behavior**
gdasgldas task is failing on the first cycle in which the task is not skipped (first 00z analysis period when enough data has been produced to trigger task).

**Machines affected**
This error is being expressed on Orion.

**To Reproduce**
I am seeing this bug in a test of global-workflow being conducted on Orion in the following directories:
expid: /work/noaa/da/bhoover/para/bth_test
code: /work/noaa/da/bhoover/global-workflow
ROTDIR: /work/noaa/stmp/bhoover/ROTDIRS/bth_test
RUNDIR: /work/noaa/stmp/bhoover/RUNDIRS/bth_test

This run is initialized on 2020082200, and designed to terminate 2 weeks later on 2020090500.

Experiment setup:
/work/noaa/da/bhoover/global-workflow/ush/rocoto/setup_expt.py --pslot bth_test --configdir  /work/noaa/da/bhoover/global-workflow/parm/config --idate 2020082200 --edate 2020090500 --comrot /work/noaa/stmp/bhoover/ROTDIRS --expdir /work/noaa/da/bhoover/para --resdet 384 --resens 192 --nens 80 --gfs_cyc 1

Workflow setup:
/work/noaa/da/bhoover/global-workflow/ush/rocoto/setup_workflow.py --expdir /work/noaa/da/bhoover/para/bth_test

Initial conditions:
/work/noaa/da/cthomas/ICS/2020082200/

The error is found in the gdasgldas task on 2020082600.

Log file:
/work/noaa/stmp/bhoover/ROTDIRS/bth_test/logs/2020082600/gdasgldas.log

**Context**
This run is being used by a new Orion user and member of the satellite DA group, only to familiarize myself with the process of carrying out an experiment. There have been no code-changes made for this run. I followed directions for cloning and building the global-workflow, and setting up a cycled experiment, from the available wiki:

https://github.com/NOAA-EMC/global-workflow/wiki/

I did not create the initial condition files, they were instead produced for me. The global-workflow repository was cloned on January 25 2022 (d3028b9d8268028226f9c27800fcd6655e9e4bb8)

The task fails with the following error in the log-file:
> 0:  NOAH Restart File Used: noah.rst                                
> 0:            1        1536         768      389408
> 0:  Restart Tile Space Mismatch, Halting..
> 0:  endrun is being called
> 0: application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0

The dimension size of 389408 is suspicious, since earlier in the log a different dimension size is referenced, e.g.:
> 0:  MSG: maketiles -- Size of Grid Dimension:      398658  (           0 )

When I search for "389408" in the log-file, it only appears in two places, one is in the Restart Tile Space Mismatch error, and the other is while running exec/gldas_rst, when reporting the results of a FAST_BYTESWAP:

> 216.121 + /work/noaa/da/bhoover/global-workflow/exec/gldas_rst
> 216.121 + 1>& 1 2>& 2
> FAST_BYTESWAP ALGORITHM HAS BEEN USED AND DATA ALIGNMENT IS CORRECT FOR          4  )
>         1536         768           4     9440776
>            2 tmp0_10cmdown GLDAS STC1
>            3 tmp10_40cmdown GLDAS STC2
>            4 tmp40_100cmdown GLDAS STC3
>            5 tmp100_200cmdown GLDAS STC4
>            6 soill0_10cmdown GLDAS SLC1
>            7 soill10_40cmdown GLDAS SLC2
>            8 soill40_100cmdown GLDAS SLC3
>            9 soill100_200cmdown GLDAS SLC4
>           10 soilw0_10cmdown GLDAS SMC1
>           11 soilw10_40cmdown GLDAS SMC2
>           12 soilw40_100cmdown GLDAS SMC3
>           13 soilw100_200cmdown GLDAS SMC4
>           15 landsfc
>           18 vtypesfc
>           71 tmpsfc GLDAS SKNT
>           72 weasdsfc GLDAS SWE
>           79 cnwatsfc GLDAS CMC
>           88 snodsfc GLDAS SNOD
>            1        1536         768      389408
> 216.602 + err=0

I believe that the error is related to the difference in tile-size between these two values.

**Detailed Description**
I have proposed no change or addition to the code for this run.

**Additional Information**
Prior gdasgldas tasks in the run from initialization to 2020082600 were successful, but they were all skipped either because the analysis was for a non-00z period or because the requisite number of cycles had not been completed to allow the task to trigger. There are no successful gdasgldas tasks in this run that I can use to compare to the one that has failed. I have conferred with more experienced EMC users of fv3gdas and the cause of the problem is not obvious.

**Possible Implementation**
I have no implementation plan to offer.

**Comment by CatherineThomas-NOAA** [comment]
@HelinWei-NOAA

**Comment by HelinWei-NOAA** [comment]
see the email from  Dave
-----------------------
Hi Helin and Jun,

I am finding with a recent upgrade of the global workflow that there is a mismatch between the land-sea mask algorithms of the UFS and the GLDAS.  This is resulting in a failure of the gdasgldas job, specifically when the Land Information System (LIS) executable is run.  The error reported is 'Restart Tile Space Mismatch, Halting..', which can be found in sorce/gldas_model.fd/lsms/noah.igbp/noahrst.F:104.  I have only verified this error for a C192/C96 run on 2020080500, where NCH = 97296 and LIS%D%NCH = 99582.

With the recent upgrade, the UFS is modifying the input land sea mask during the first gdasfcst and outputting this modified mask to the tiled surface restart files.  Thus, all future forecasts, analyses, etc use this modified mask until the GLDAS reads its own from $FIX/fix_gldas/FIX_T382/lmask_gfs_T382.bfsa.

So I have a few questions.  First, the UFS's modification of the land-sea mask is expected, correct?  Secondly, should a new fix file be created for the GLDAS with the modified land-sea mask or is the UFS-modified land-sea mask time dependent and thus not fixed?  Lastly, should I expect this to be an issue at all resolutions?

------------------------

Since GLDAS will not be included in the next operational implementation, we need to have someone to decide if we should spend more time on this task.

**Comment by yangfanglin** [comment]
Is the cycling working without gdasgldas ?  Have the user tried at the operational resolutions (C768/C384) ?

**Comment by BrettHoover-NOAA** [comment]
The cycling appears to fail when gdasgldas fails. I have not run at
operational resolution, the current run is an initial test-run I'm doing as
a new global-workflow user on Orion and the initial conditions were
provided to me at C384/C192.

Brett

**Comment by CatherineThomas-NOAA** [comment]
@yangfanglin The experiment had no errors when running in the first few days before GLDAS turns on, so I assume that it would run if GLDAS was turned off altogether.  

If as @HelinWei-NOAA says that GLDAS will not be used going forward, should we turn it off now in our experiments?  The DA team is still running atmosphere-only cases since we have non-coupling data upgrades to worry about.

**Comment by yangfanglin** [comment]
@CatherineThomas-NOAA If the cycling at C384/C192 resolutions with gdasgldas turned on is working on WCOSS, there is likely issues related to the setups on Orion.  I had some discussion with Daryl about the use of gdasgldas in future systems. We can discuss more about this issue with Daryl offline.

**Comment by CatherineThomas-NOAA** [comment]
@AndrewEichmann-NOAA Has your WCOSS experiment progressed to the point where the gldas step is run?  If so, did it fail or run without issue?

**Comment by AndrewEichmann-NOAA** [comment]
@CatherineThomas-NOAA No - I ran into rstprod access issues and am waiting for a response from helpdesk

**Comment by CatherineThomas-NOAA** [comment]
Thanks @AndrewEichmann-NOAA.  I can run a short warm start test on WCOSS.

**Comment by CatherineThomas-NOAA** [comment]
The warm start test on WCOSS ran the gldas step without failure.  Now that Hera is back, I can try a quick test there as well.

**Comment by CatherineThomas-NOAA** [comment]
> Now that Hera is back, I can try a quick test there as well.

Global-workflow develop does not build on Hera (#561), so I can't run this test at this time.

**Comment by DavidHuber-NOAA** [comment]
I ran into this issue with the S4 and Jet ports, which I reported to Helin.  I have since turned off GLDAS altogether and everything has run OK out to 10 days.

Below is more of the thread between Helin, @junwang-noaa, and myself:

David,

GLDAS should use the same land-sea mask as UFS. If the land-sea mask can be changed during the forecast, that certainly will bring up some issues for GLDAS. 

Helin

Dave,

The model does check/modify the land-sea mask according to the input land/lake fraction and soil type values, this is applied to both non-fractional grid and fractional grid. The changes are required to make sure the model has consistent land sea mask, and soil types. The new land sea mask is output in the model sfc file. I expect you do not change the oro data, the soil type data during your run, so this land sea mask won't change. In other words, I think you need to create lmask_gfs_T382.bfsa once with this land sea mask in the history file.

Jun

**Comment by arunchawla-NOAA** [comment]
@HelinWei-NOAA @barlage and @yangfanglin should Jun's suggestion be followed at the mask be generated for GLDAS from the history file ? It would be good to know if the same issue is seen on other platforms

**Comment by yangfanglin** [comment]
@arunchawla-NOAA   Cathy reported that "the warm start test on WCOSS ran the gldas step without failure".  So the failures on other platforms might be a porting issue.  @CatherineThomas-NOAA Cathy, can you confirm ? What is the  resolution you tested on WCOSS ?  I assume you were using the GFS.v16* tag instead of the UFS, right ?

**Comment by WalterKolczynski-NOAA** [comment]
@BrettHoover-NOAA I'm unable to access either the code or experiment directories.

I ran a test on Orion a few days ago and didn't have any issue, so this probably isn't a port issue. I'm setting up another test just to be sure.

**Comment by CatherineThomas-NOAA** [comment]
My test on WCOSS was C384/C192 with warm start initial conditions from the PR [#500](https://github.com/NOAA-EMC/global-workflow/pull/500) test.  This was using the head of develop global-workflow at the time [(d3028b9),](https://github.com/NOAA-EMC/global-workflow/commit/d3028b9d8268028226f9c27800fcd6655e9e4bb8) compiling and running with atmosphere only.  Since then, I've run new tests on Hera and Orion with the recent update to develop [(97ebc4d)](https://github.com/NOAA-EMC/global-workflow/commit/97ebc4d4e6483b135468d85cd9ed974f88955914) and ran into no issues with gldas.

@BrettHoover-NOAA It's possible this issue got fixed inadvertently with the recent develop update.  It could also be related to the set of ICs that you started from.  How about you try to replicate the test I ran first?  I'll point you to my initial conditions offline.

**Comment by WalterKolczynski-NOAA** [comment]
I was able to run a 6½-cycle warm-start from a fresh clone overnight without issue. I'm also using the C384 ICs Cathy produced for PR #500.

**Comment by BrettHoover-NOAA** [comment]
@CatherineThomas-NOAA I was able to complete your test with the new develop [(97ebc4d)](https://github.com/NOAA-EMC/global-workflow/commit/97ebc4d4e6483b135468d85cd9ed974f88955914) and warm-start ICs on Orion, and I ran into no problems, gdasgldas appears to finish successfully.

**Comment by CatherineThomas-NOAA** [comment]
@BrettHoover-NOAA Great to hear.  It looks like your Orion environment is working properly.  Maybe to round out this set of tests you could try the warm-start 2020083000 ICs but with the original workflow that you cloned, assuming you still have it.

**Comment by BrettHoover-NOAA** [comment]
@CatherineThomas-NOAA I have that test running right now, I'll report back ASAP

**Comment by BrettHoover-NOAA** [comment]
@CatherineThomas-NOAA The warm-start test with the original workflow also finished successfully.

**Comment by CatherineThomas-NOAA** [comment]
@BrettHoover-NOAA Great!  There may have been an incompatibility with the other ICs then.  

@HelinWei-NOAA @DavidHuber-NOAA Is the land-sea mask problem that you mentioned early documented elsewhere?  Can this issue be closed?

**Comment by HelinWei-NOAA** [comment]
@CatherineThomas-NOAA No. It  hasn't been documented elsewhere. But I have let Fanglin and Mike know this issue. IMO this issue can be closed now.
> @BrettHoover-NOAA Great! There may have been an incompatibility with the other ICs then.
> 
> @HelinWei-NOAA @DavidHuber-NOAA Is the land-sea mask problem that you mentioned early documented elsewhere? Can this issue be closed?

**Comment by DavidHuber-NOAA** [comment]
@CatherineThomas-NOAA I'm also OK with this issue being closed.

**Comment by DavidHuber-NOAA** [comment]
I gave this a fresh cold start test (C192/C96) on Orion over the weekend and received the same error.  Initial conditions were generated on Hera (/scratch1/NESDIS/nesdis-rdo2/David.Huber/ufs_utils/util/gdas_init), outputting them here: /scratch1/NESDIS/nesdis-rdo2/David.Huber/output/192.  UFS_Utils was checked out using the same hash as the global workflow checkout script ([04ad17e2](https://github.com/ufs-community/UFS_UTILS/commit/04ad17e27163b93a53ba3bfdbf38fd0ea82d9b1b)).

These were then transferred to Orion where a test ran from 2020073118 through 2020080500, where gdasgldas failed with the same message ("Restart Tile Space Mismatch, Halting..").  The global workflow hash used was [64b1c1e](https://github.com/NOAA-EMC/global-workflow/commit/64b1c1e5ce37fba48dd717bd11356a6f57d9def4) and can be found here: /work/noaa/nesdis-rdo2/dhuber/gw_dev.  Logs from the run can be found here: /work/noaa/nesdis-rdo2/dhuber/para/com/test_gldas/logs.

A comparison of the land surface mask (`lsmsk`) between the IC tile1 surface file and the tile1 restart file shows a difference.

I also created initial conditions for C384/C192 and compared the land surface mask against Cathy's tile 1 restart surface file, which shows no difference.

Lastly, I copied the C384/C192 ICs over to Orion and executed just the gdasfcst job, then compared the `lsmsk` field as before and there was a difference.

It is this modification by the UFS that triggers problems for GLDAS and I think any further tests could be limited to just a single half-cycle run of gdasfcst.  I tracked this modification to the [addLsmask2grid subroutine](https://github.com/NOAA-EMC/fv3atm/blob/8afe51e12a6f93b022bf1d2cf97b4d0784748f7f/atmos_model.F90#L2819-L2880), which is tied to ocean fraction which in turn is set in the orographic fix files, which are identical on Orion and Hera.  So I am at a loss as to why these differ between warm and cold starts.  Is this expected behavior and if so should GLDAS be turned off for cold starts?

**Comment by KateFriedman-NOAA** [comment]
All, is there guidance on whether users should turn off GLDAS when running cold-started experiments for now? @jkhender just hit the same error on Orion with a cold-started experiment using global-workflow `develop` as of 2/2/22. Thanks!

**Comment by jkhender** [comment]
correction - my experiment is running on Hera

**Comment by DavidHuber-NOAA** [comment]
This is what I would suggest.  I don't think that a fix file update will work for everyone since warm starts seem to be using the current fix files without a problem, implying that an update would result in a mismatch for those users (testing could confirm this).  Alternatively, two sets of fix files could be created, one for warm starts and one for cold, but that would require some scripting to know which to link to.

**Comment by KateFriedman-NOAA** [comment]
> correction - my experiment is running on Hera

Oops, thanks for correcting @jkhender !

> Alternatively, two sets of fix files could be created, one for warm starts and one for cold, but that would require some scripting to know which to link to.

So this is related to a fix file issue? Sorry, been out of the loop on this. Thanks!

**Comment by DavidHuber-NOAA** [comment]
> So this is related to a fix file issue? Sorry, been out of the loop on this. Thanks!

I believe so, yes.  See the email thread between Jun, Helin, and myself [here ](https://github.com/NOAA-EMC/global-workflow/issues/622#issuecomment-1026326625) and [here](https://github.com/NOAA-EMC/global-workflow/issues/622#issuecomment-1028278324).

**Comment by AndrewEichmann-NOAA** [comment]
I've now encountered this on WCOSS, on the fourth full 00Z cycle into a run. I modified the rocoto script to skip over it. 

rotdir /gpfs/dell1/ptmp/Andrew.Eichmann/baily
expdir /gpfs/dell2/emc/modeling/save/Andrew.Eichmann/para/baily

global-workflow is my EFOSI fork branch off from master
d7319f19aceca6ae6d7ce9b06c6eb731832d1de1 (Feb 2)

**Comment by KateFriedman-NOAA** [comment]
@AndrewEichmann-NOAA For clarification...you skipped the failing GLDAS job in your fourth full 00z cycle but kept GLDAS on for following cycles? If so, did the GLDAS work in the next cycles? Thanks!

**Comment by AndrewEichmann-NOAA** [comment]
@KateFriedman-NOAA I skipped the fourth full 00z cycle, tried turning off in config.base, which didn't have any effect (still in the workflow), so I regenerated the xml file to have the workflow without it (this being since I wrote the above). Basically I tried not running it at all, so I don't know if it would have worked properly on later cycles.

**Comment by KateFriedman-NOAA** [comment]
@AndrewEichmann-NOAA Gotcha, thanks for the clarification!

All, it sounds like we should advise that users don't run with GLDAS if they are cold-starting an experiment, especially given the note from @HelinWei-NOAA that we won't be using the GLDAS moving forward. @yangfanglin @CatherineThomas-NOAA @WalterKolczynski-NOAA Any objections to this guidance? We have users with stalled experiments because of this cold-start problem. Thanks!

**Comment by CatherineThomas-NOAA** [comment]
@KateFriedman-NOAA No objection from me.

**Comment by RussTreadon-NOAA** [comment]
What guidance do we have for developers who cold start and experiment AND want to run GLDAS?    The JEDI GDAS wants to mimic all operational job steps apart from the analysis step which will be JEDI VAR.    As this is just a prototype we are running at C96L127.    After cycling sufficient days to run the 00Z GLDAS, job gdasgldas fails with the restart tile space mismatch error reported in this issue.

Is there an offline executable we can run to make a cold start experiment look like a warm start experiment?   What is the target date for turning off the GLDAS in operations?    I don't view turning off GLDAS as a solution.

**Comment by RussTreadon-NOAA** [comment]
As requested by g-w staff, this issue has been reopened.

**Comment by RussTreadon-NOAA** [comment]
An Orion C96L127 parallel running ufs model tag `release/P8a` was used to create a test set of `fix/fix_gldas/FIX_T190` files.   

A 00Z gdasgldas job using `/work/noaa/global/glopara/fix_NEW/fix_gldas/FIX_T190` was submitted.  This job failed with the error `Restart Tile Space Mismatch`.    The test `FIX_T190` files were swapped in and the 00Z gdasgldas job rerun.  The job ran to completion without any error messages.

**Comment by yangfanglin** [comment]
@RussTreadon-NOAA , what are the differences between the original FIX_T190 files  and test FIX_T190 files  ?  Sorry if it has been discussed and document somewhere else.

**Comment by RussTreadon-NOAA** [comment]
@HelinWei-NOAA created the FIX_T190 files using output from the Orion C96L127 parallel.

**Comment by yangfanglin** [comment]
@HelinWei-NOAA Thanks Helin for creating the new files.  Are you planning to create the new fix files for other resolutions as well ?

**Comment by HelinWei-NOAA** [comment]
@yangfanglin Yes. As long as the sfc nemsio files for other resolutions can be provided.

**Comment by HelinWei-NOAA** [comment]
@KateFriedman-NOAA My plan is to add a flag "GLDAS_FIX" to config.base, it is set to UFS or GFS. 

### run GLDAS to spin up land ICs
export DO_GLDAS="YES"
export gldas_cyc=00export  GLDAS_FIX="UFS"
if [ $GLDAS_FIX = "UFS" ]; then
export FIXgldas=the place of the new gldas fixed fields based on UFS
fi

What do you think?Do you prefer to move this section to config.gldas?

**Comment by KateFriedman-NOAA** [comment]
@HelinWei-NOAA I would prefer GLDAS-specific settings go into config.gldas. Perhaps `gldas_cyc` can go in there too. @WalterKolczynski-NOAA thoughts?

Questions:

1) Where will these UFS-FIXgldas files live? Can we add them into the main `FIX_DIR` alongside the GFS-FIXgldas files? Can you explain how they will differ between GFS and UFS? We currently have the following for GLDAS in `FIX_DIR`:
```
[Kate.Friedman@v71a1 fix_gldas]$ pwd
/gpfs/dell2/emc/modeling/noscrub/emc.glopara/git/fv3gfs/fix_NEW/fix_gldas
[Kate.Friedman@v71a1 fix_gldas]$ ll
total 16
drwxr-xr-x 2 emc.glopara emcmodel 2048 Dec 13  2019 FIX_T1534
drwxr-xr-x 2 emc.glopara emcmodel 2048 Dec 13  2019 FIX_T190
drwxr-xr-x 2 emc.glopara emcmodel 2048 Dec 13  2019 FIX_T382
drwxr-xr-x 2 emc.glopara emcmodel 2048 Dec 13  2019 FIX_T766
```
2) How will we/users know when to switch to the UFS set instead of the GFS set? Can we use another switch to determine that? Is this UFS set for the coupled system? If so, perhaps it could key off of the `APP` setting?

**Comment by HelinWei-NOAA** [comment]
@WalterKolczynski-NOAA @KateFriedman-NOAA I am very confused about two directories under global-workflow/fix. fix_fv3_fracoro is a copy of fix_fv3_gmted2010 but with the latest data used by UFS PT. How does the workflow determine which one will be used? We should only use one of them for consistency. But now it seems to me  that FIX_SFC is linked to fix_fv3_fracoro and FIXfv3 is linked to fix_fv3_gmted2010.

**Comment by KateFriedman-NOAA** [comment]
> @WalterKolczynski-NOAA @KateFriedman-NOAA I am very confused about two directories under global-workflow/fix. fix_fv3_fracoro is a copy of fix_fv3_gmted2010 but with the latest data used by UFS PT. How does the workflow determine which one will be used? We should only use one of them for consistency. But now it seems to me that FIX_SFC is linked to fix_fv3_fracoro and FIXfv3 is linked to fix_fv3_gmted2010.

@HelinWei-NOAA The following forecast job child scripts set the mentioned FIX paths:

* `ush/forecast_predet.sh` sets `FIXfv3` as `FIXfv3=${FIXfv3:-$FIX_DIR/fix_fv3_gmted2010}`
* `ush/forecast_postdet.sh` sets `OROFIX` and `FIX_SFC` depending on the value of `FRAC_GRID`:
```
  # Fractional grid related
  if [ $FRAC_GRID = ".true." ]; then
    OROFIX=${OROFIX:-"${FIX_DIR}/fix_fv3_fracoro/${CASE}.mx${OCNRES}_frac"}
    FIX_SFC=${FIX_SFC:-"${OROFIX}/fix_sfc"}
    for n in $(seq 1 $ntiles); do
      $NLN ${OROFIX}/oro_${CASE}.mx${OCNRES}.tile${n}.nc $DATA/INPUT/oro_data.tile${n}.nc
    done
  else
    OROFIX=${OROFIX:-"${FIXfv3}/${CASE}"}
    FIX_SFC=${FIX_SFC:-"${FIXgrd}/fix_sfc"}
    for n in $(seq 1 $ntiles); do
      $NLN ${OROFIX}/${CASE}_oro_data.tile${n}.nc $DATA/INPUT/oro_data.tile${n}.nc
    done
  fi
```

Tagging @yangfanglin to explain fix file usage further and check we are being consistent/correct.

**Comment by JessicaMeixner-NOAA** [comment]
The fix_fv3_fracoro should not just simply be a copy of the fix_fv3_gmted2010.  These have the fractional masks (ie. a grid cell can have both ocean and land).  @shansun6 and others added the capability for fractional grid and would likely be the best person to provide more details on the contents of the fracoro directory.

**Comment by HelinWei-NOAA** [comment]
@KateFriedman-NOAA Thanks for your explanation. @yangfanglin In the current setting, the new dataset including viirs-based vegetation type can only be used when fractional grid is used. I am wondering if they should be also used (if they can @shansun6) without fractional grid. You turn fractional grid off in your cycled test run, but gldas still failed. We need to figure this out.

**Comment by HelinWei-NOAA** [comment]
@JessicaMeixner-NOAA When I said a copy, I meant the structure is the same (number of files, file names). The only difference is the oro data has more information (lake fracyion, lake depth) and all other fixed fields are based on this oro data.

**Comment by HelinWei-NOAA** [comment]
@shansun6 If the dataset created for fractional grid can't be used for the run without fractional grid, probably you should create two datasets w/o fractional grid from the same raw data at the same time for consistency.

**Comment by HelinWei-NOAA** [comment]
@yangfanglin Should the oro data under the fix_ugwd used by the gravity drag scheme be also updated when we have the new oro data?

**Comment by ShanSunNOAA** [comment]
@HelinWei-NOAA Any details describing the nature of the crash? Would be good to know the location of the crash: is it an ocean coast or inland lake point? 

I can provide an non-fractional version of the oro_data to be used solely as a debugging tool. For the production mode, it is a good practice to stay with one unique oro_data to be consistent. 

On the other hand, with some code changes in FV3 io routine, dataset created for fractional grid can be used to run FV3 in the non-fractional mode, as the fractional dataset contains additional info of the non-fractional grid has.

Let me know how you'd like to proceed.

**Comment by yangfanglin** [comment]
> @yangfanglin Should the oro data under the fix_ugwd used by the gravity drag scheme be also updated when we have the new oro data?

@HelinWei-NOAA

**Comment by yangfanglin** [comment]
> > @yangfanglin Should the oro data under the fix_ugwd used by the gravity drag scheme be also updated when we have the new oro data?
> 
> @HelinWei-NOAA

fix_ugwd works for both fractional and non_fractional grid.   It needn't be updated if oro_data is changed since it was created by   Mike Toy using a different oro dataset.

**Comment by yangfanglin** [comment]
> 

@HelinWei-NOAA My cycled run without fractional grid completed the gdasgldas step after pointing to the fix_gldas you updated.  I did not the experiment beyond the gdasgldas step.  There might be some mismatches.

**Comment by RussTreadon-NOAA** [comment]
Reopen after consulting with @yangfanglin .   This issue remains to be resolved.

**Comment by HelinWei-NOAA** [comment]
Okay. Here is my plan. I will create two datasets. One is for FRAC_GRID = ".true." and the other one is for FRAC_GRID = ".false.".  The condition will be added to ush/forecast_predet.sh. The previos gldas failure without fractional grid is because the old GLDAS fixed fields were created based on the Gaussian fixed files but coldstarting ufs cycled run uses titled sfc fixed files. There are some mismatches.

Now I already have the new GLDAS fixed fields for these resolutions:

FRAC_GRID = ".false.": FIX_T190  FIX_T382  FIX_T766
FRAC_GRID = ".true.": FIX_T190

@DavidHuber-NOAA Can you run other cycled ufs tests with gldas for me?

C768 with FRAC_GRID = ".false."
C768, C384, C192 with FRAC_GRID = ".true."

> > @WalterKolczynski-NOAA @KateFriedman-NOAA I am very confused about two directories under global-workflow/fix. fix_fv3_fracoro is a copy of fix_fv3_gmted2010 but with the latest data used by UFS PT. How does the workflow determine which one will be used? We should only use one of them for consistency. But now it seems to me that FIX_SFC is linked to fix_fv3_fracoro and FIXfv3 is linked to fix_fv3_gmted2010.
> 
> @HelinWei-NOAA The following forecast job child scripts set the mentioned FIX paths:
> 
> * `ush/forecast_predet.sh` sets `FIXfv3` as `FIXfv3=${FIXfv3:-$FIX_DIR/fix_fv3_gmted2010}`
> * `ush/forecast_postdet.sh` sets `OROFIX` and `FIX_SFC` depending on the value of `FRAC_GRID`:
> 
> ```
>   # Fractional grid related
>   if [ $FRAC_GRID = ".true." ]; then
>     OROFIX=${OROFIX:-"${FIX_DIR}/fix_fv3_fracoro/${CASE}.mx${OCNRES}_frac"}
>     FIX_SFC=${FIX_SFC:-"${OROFIX}/fix_sfc"}
>     for n in $(seq 1 $ntiles); do
>       $NLN ${OROFIX}/oro_${CASE}.mx${OCNRES}.tile${n}.nc $DATA/INPUT/oro_data.tile${n}.nc
>     done
>   else
>     OROFIX=${OROFIX:-"${FIXfv3}/${CASE}"}
>     FIX_SFC=${FIX_SFC:-"${FIXgrd}/fix_sfc"}
>     for n in $(seq 1 $ntiles); do
>       $NLN ${OROFIX}/${CASE}_oro_data.tile${n}.nc $DATA/INPUT/oro_data.tile${n}.nc
>     done
>   fi
> ```
> 
> Tagging @yangfanglin to explain fix file usage further and check we are being consistent/correct.

**Comment by KateFriedman-NOAA** [comment]
> @DavidHuber-NOAA Can you run other cycled ufs tests with gldas for me?

@HelinWei-NOAA Apologies for the delayed response to this! FYI, Dave is currently out on medical leave so he's not available to assist with testing right now.

> Okay. Here is my plan. I will create two datasets. One is for FRAC_GRID = ".true." and the other one is for FRAC_GRID = ".false.". The condition will be added to ush/forecast_predet.sh. The previos gldas failure without fractional grid is because the old GLDAS fixed fields were created based on the Gaussian fixed files but coldstarting ufs cycled run uses titled sfc fixed files. There are some mismatches.
> Now I already have the new GLDAS fixed fields for these resolutions:
> FRAC_GRID = ".false.": FIX_T190 FIX_T382 FIX_T766
> FRAC_GRID = ".true.": FIX_T190

This sounds good to me, just need to test it. @aerorahul Is there another developer that can assist with this? Thanks!

**Comment by DavidHuber-NOAA** [comment]
> @DavidHuber-NOAA Can you run other cycled ufs tests with gldas for me?
> 
> C768 with FRAC_GRID = ".false."
> C768, C384, C192 with FRAC_GRID = ".true."

@HelinWei-NOAA I'm ready to start gathering the new data.  Just to confirm, you only need me to run the first, cold start forecast with the UFS at the above resolutions and provide the sfc files, correct?  Also, do you need these output in nemsio?  Lastly, if you already have the FIX_T766 file for FRAC_GRID="false", does that case need to be rerun still?

**Comment by HelinWei-NOAA** [comment]
@DavidHuber-NOAA With updates on some of fixed fields (oro, land-sea mask), I would like to regenerate all fixed fields for GLDAS. I only need one sfc file in nemsio (something like sfc.gaussian.nemsio) for each case. Thanks.

**Comment by DavidHuber-NOAA** [comment]
Just a note, for anyone wishing to run with GLDAS enabled, `FHOUT` must be set to 1 in config.base.  To speed things up, I turned IAU and 4DVAR off, which disabled hourly outputs.  A conditional should probably be added to the configuration file to change `FHOUT=1` if `DO_GLDAS=YES`.

**Comment by DavidHuber-NOAA** [comment]
@HelinWei-NOAA The surface files have now been generated.  You can find them on Hera here: /scratch1/NESDIS/nesdis-rdo2/David.Huber/for_helin/sfc_files.tar.

**Comment by KateFriedman-NOAA** [comment]
@HelinWei-NOAA @DavidHuber-NOAA Where are we with this? I'd like to get GLDAS working again if possible. Thanks!

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA The data requested by @HelinWei-NOAA to generate the new fix files was generated and placed on Hera.  I believe these just need to be parsed by Helin.

**Comment by KateFriedman-NOAA** [comment]
@HelinWei-NOAA Any updates on parsing the provided new fix files? We'd like to get this issue resolved. Thanks!

**Comment by HelinWei-NOAA** [comment]
Kate,

I don't have the new fixed fields for all resolutions. I will provide you with what I have.

**Comment by KateFriedman-NOAA** [comment]
> I don't have the new fixed fields for all resolutions. I will provide you with what I have.

Ok, thanks!

**Comment by HelinWei-NOAA** [comment]
@DavidHuber-NOAA My job failed when I processed the data you created, I found the size of those nemsio files are not right. For example the size of /scratch1/NCEPDEV/global/glopara/fix/gldas/20220805/FIX_T766/sfc.gaussian.nemsio.t766.1536.768 is 434114776, but yours is 1736446168 (almost 4 times larger). Can you double check if it is one time output?

> /scratch1/NESDIS/nesdis-rdo2/David.Huber/for_helin/sfc_files.tar

**Comment by DavidHuber-NOAA** [comment]
@HelinWei-NOAA Looking into the files with nemsio_read I can see that the number of grid points is 4x the model resolution.  For instance, the `C192` data creates the `T382`-sized grid.  So it looks like you should be matching the data as follows:

192/sfc.gaussian.nemsio.20200802 -> fix/gldas/20220805/FIX_T382/sfc.gaussian.nemsio.t382.768.384
384/sfc.gaussian.nemsio.20200802 -> fix/gldas/20220805/FIX_T766/sfc.gaussian.nemsio.t766.1536.768
768/sfc.gaussian.nemsio.20200802 -> fix/gldas/20220805/FIX_T1534/sfc.gaussian.nemsio.t1534.3027.1536
768_false/sfc.gaussian.nemsio.20200802 -> fix/gldas/20220805/FIX_T1534/sfc.gaussian.nemsio.t1534.3027.1536

That also means that a file was not generated for FIX_T192.  I can do this pretty quickly, though, and I will add it to the same tarball when complete.

**Comment by HelinWei-NOAA** [comment]
@DavidHuber-NOAA Thanks for quick turn around. It would be great if you can create one for FIX_T192 and another one for T1534 for frac_grid=off. Then we have full set of data.

**Comment by DavidHuber-NOAA** [comment]
@HelinWei-NOAA Alright, I'll start on FIX_T192.  The T1534 for frac_grid=off is already complete and is in the 768_false directory.

**Comment by HelinWei-NOAA** [comment]
@DavidHuber-NOAA Yes. We already had that. I also found we had FIX_T192 for frac_grid=on too. You don't need to create again. Thanks.

**Comment by DavidHuber-NOAA** [comment]
@HelinWei-NOAA Great!  Let me know if you need anything else.

**Comment by HelinWei-NOAA** [comment]
@KateFriedman-NOAA The fixed fields for GLDAS are ready. You can copy data from /scratch1/NCEPDEV/global/Helin.Wei/save/fix_gldas. When frac_grid is on, please point fix_gldas to  fix_gldas/frac_grid. When frac_grid is off, please point to fix_gldas/nonfrac_grid.

**Comment by KateFriedman-NOAA** [comment]
> The fixed fields for GLDAS are ready.

Thanks @HelinWei-NOAA ! I'll try them out and report back.

**Comment by KateFriedman-NOAA** [comment]
@HelinWei-NOAA Can you adjust the permissions on the following file?

/scratch1/NCEPDEV/global/Helin.Wei/save/fix_gldas/frac_grid/FIX_T190/sfc.gaussian.nemsio.t190.384.192

I am getting "Permission denied" trying to copy the updated set to Orion for testing. I would guess I'd get a similar runtime issue if I tested on Hera. Please also double check the other fix_gldas file permissions to make sure they are what is needed. Thanks!

**Comment by HelinWei-NOAA** [comment]
They should be fine now.

**Comment by KateFriedman-NOAA** [comment]
Yep, I was able to copy that file to Orion, thanks @HelinWei-NOAA !

**Comment by KateFriedman-NOAA** [comment]
@HelinWei-NOAA Using the `frac_grid` version of the gldas fix files I was able to get a previously-failing GLDAS job to complete without additional failures. Please see the following log on Orion and confirm if the job ran correctly:

/work/noaa/stmp/kfriedma/comrot/testgldas/logs/2022010600/gdasgldas.log

Thanks!

**Comment by HelinWei-NOAA** [comment]
@KateFriedman-NOAA Good to know. Thanks for testing.

**Comment by KateFriedman-NOAA** [comment]
The following GLDAS script updates are necessary to use the updated GLDAS fix set that supports both `FRAC_GRID` on or off:

```
Orion-login-1[264] /work/noaa/global/kfriedma/git/develop_gldas$ git diff
diff --git a/jobs/JGDAS_ATMOS_GLDAS b/jobs/JGDAS_ATMOS_GLDAS
index 62196154..e83fea14 100755
--- a/jobs/JGDAS_ATMOS_GLDAS
+++ b/jobs/JGDAS_ATMOS_GLDAS
@@ -8,7 +8,7 @@ export RUN_ENVIR=${RUN_ENVIR:-"nco"}
 # Source relevant config files
 #############################
 export EXPDIR=${EXPDIR:-${HOMEgfs}/parm/config}
-configs="base gldas"
+configs="base fcst gldas"
 config_path=${EXPDIR:-${NWROOT}/gfs.${gfs_ver}/parm/config}
 for config in ${configs}; do
     # shellcheck source=parm/config/config.${config}
diff --git a/scripts/exgdas_atmos_gldas.sh b/scripts/exgdas_atmos_gldas.sh
index 2bd0154c..742ca969 100755
--- a/scripts/exgdas_atmos_gldas.sh
+++ b/scripts/exgdas_atmos_gldas.sh
@@ -60,7 +60,7 @@ nlat=$((2*res))
 nlon=$((4*res))

 export USHgldas=${USHgldas:?}
-export FIXgldas=${FIXgldas:-${HOMEgfs}/fix}
+export FIXgldas=${FIXgldas:-${HOMEgfs}/fix/fix_gldas}
 export topodir=${topodir:-${HOMEgfs}/fix/fix_fv3_gmted2010/${CASE}/}

 DATA=${DATA:-${pwd}/gldastmp$$}
@@ -84,9 +84,14 @@ input2=${COMINgdas}/gdas.${gldas_eymd}/${gldas_ecyc}/${COMPONENT}/RESTART
 [[ -d ${RUNDIR}/input ]] && rm -fr "${RUNDIR}/input"
 mkdir -p "${RUNDIR}/input"
 ln -fs "${GDAS}" "${RUNDIR}/input/GDAS"
-ln -fs "${FIXgldas}/FIX_T${JCAP}" "${RUNDIR}/FIX"
 ln -fs "${EXECgldas:?}/gldas_model" "${RUNDIR}/LIS"

+# Set FIXgldas subfolder based on FRAC_GRID value
+if [[ "${FRAC_GRID:-".true."}" = ".true." ]] ; then
+  ln -fs "${FIXgldas}/frac_grid/FIX_T${JCAP}" "${RUNDIR}/FIX"
+else
+  ln -fs "${FIXgldas}/nonfrac_grid/FIX_T${JCAP}" "${RUNDIR}/FIX"
+fi
```

Summary of changes:

1. Add sourcing of `config.fcst` to get the value of `FRAC_GRID` (.true. or not) for use in the GLDAS
2. Update `scripts/exgdas_atmos_gldas.sh` to check `FRAC_GRID` value and set `FIX` symlink in rundir based on the appropriate subfolder of `FIXgldas`: use `$FIXgldas/frac_grid` when `FRAC_GRID=.true.` or use `$FIXgldas/nonfrac_grid` when `FRAC_GRID=.false.`
3. Also included a small fix to the `FIXgldas` default value, it was missing the "/fix_gldas" part:
`export FIXgldas=${FIXgldas:-${HOMEgfs}/fix_gldas}`

Note: the edits above are shown in the copy of `scripts/exgdas_atmos_gldas.sh` within my global-workflow branch that absorbs and updates the GLDAS scripts to correct linter/shellcheck warnings.

**Comment by KateFriedman-NOAA** [comment]
Moving `FRAC_GRID` setting from `config.fcst` to `config.base.*` since the value is now used in more than just the forecast job. Updated diff:
```
Orion-login-1[334] /work/noaa/global/kfriedma/git/develop_gldas$ git diff
diff --git a/parm/config/config.base.emc.dyn b/parm/config/config.base.emc.dyn
index af18173d..e0168205 100755
--- a/parm/config/config.base.emc.dyn
+++ b/parm/config/config.base.emc.dyn
@@ -162,6 +162,7 @@ export CCPP_SUITE="FV3_GFS_v17_p8"
 export WAVE_CDUMP="" # When to include wave suite: gdas, gfs, or both
 export DOBNDPNT_WAVE="NO"
 export cplwav2atm=".false."
+export FRAC_GRID=".true."

 case "${APP}" in
   ATM)
diff --git a/parm/config/config.fcst b/parm/config/config.fcst
index cb1add3f..51bee1e2 100755
--- a/parm/config/config.fcst
+++ b/parm/config/config.fcst
@@ -252,7 +252,6 @@ export DO_OCN_SPPT=${DO_OCN_SPPT:-"NO"}
 export DO_OCN_PERT_EPBL=${DO_OCN_PERT_EPBL:-"NO"}

 #coupling settings
-export FRAC_GRID=".true."
 export cplmode="nems_frac"
 if [ $FRAC_GRID = ".false." ]; then
   export cplmode="nems_orig"
diff --git a/scripts/exgdas_atmos_gldas.sh b/scripts/exgdas_atmos_gldas.sh
index 2bd0154c..4cae6f83 100755
--- a/scripts/exgdas_atmos_gldas.sh
+++ b/scripts/exgdas_atmos_gldas.sh
@@ -60,7 +60,7 @@ nlat=$((2*res))
 nlon=$((4*res))

 export USHgldas=${USHgldas:?}
-export FIXgldas=${FIXgldas:-${HOMEgfs}/fix}
+export FIXgldas=${FIXgldas:-${HOMEgfs}/fix/fix_gldas}
 export topodir=${topodir:-${HOMEgfs}/fix/fix_fv3_gmted2010/${CASE}/}

 DATA=${DATA:-${pwd}/gldastmp$$}
@@ -84,9 +84,14 @@ input2=${COMINgdas}/gdas.${gldas_eymd}/${gldas_ecyc}/${COMPONENT}/RESTART
 [[ -d ${RUNDIR}/input ]] && rm -fr "${RUNDIR}/input"
 mkdir -p "${RUNDIR}/input"
 ln -fs "${GDAS}" "${RUNDIR}/input/GDAS"
-ln -fs "${FIXgldas}/FIX_T${JCAP}" "${RUNDIR}/FIX"
 ln -fs "${EXECgldas:?}/gldas_model" "${RUNDIR}/LIS"

+# Set FIXgldas subfolder based on FRAC_GRID value
+if [[ "${FRAC_GRID:-".true."}" = ".true." ]] ; then
+  ln -fs "${FIXgldas}/frac_grid/FIX_T${JCAP}" "${RUNDIR}/FIX"
+else
+  ln -fs "${FIXgldas}/nonfrac_grid/FIX_T${JCAP}" "${RUNDIR}/FIX"
+fi
```

**Comment by KateFriedman-NOAA** [comment]
Ran an additional four cycles of both control and test run on Orion. The next 00z cycle invoked GLDAS again and correctly ran in both runs. Both the control and test run continued to reproduce each other. Changes to use updated fix files looks good.

Committed above script updates to `feature/gldas_scripts` issue #1014 branch @ https://github.com/NOAA-EMC/global-workflow/pull/1018/commits/9a6a112f583fa794b4b45681c7d530dabb2ab829.

**Comment by KateFriedman-NOAA** [comment]
@HelinWei-NOAA @DavidHuber-NOAA I tried the updated GLDAS fix files (`frac_grid` subset with `FRAC_GRID=.true.`) on WCOSS2 in an on-going high-res long-ago warm-started test and the first gldas job to use the new set (with companion script updates) failed with the same mismatch error:
```
nid001611.cactus.wcoss2.ncep.noaa.gov 0:  NOAH Restart File Used: noah.rst
nid001611.cactus.wcoss2.ncep.noaa.gov 0:            1        3072        1536     1593676
nid001611.cactus.wcoss2.ncep.noaa.gov 0:  Restart Tile Space Mismatch, Halting..
 endrun is being called
```

The prior cycles in this high-res test ran the gldas job successfully using existing (not new) fix set and was warm-started with ops output 15 days ago. Prior cycles used the older/existing fix set:

Hera copy: /scratch1/NCEPDEV/global/glopara/fix/gldas/20220805/FIX_T1534

...whereas this failing gldas job used the updated set:

Hera copy: /scratch1/NCEPDEV/global/glopara/fix/gldas/20220920/frac_grid/FIX_T1534

^ those Hera copies are both mirrored onto WCOSS2 but since Cactus is down today and I'm not sure if @DavidHuber-NOAA has access to WCOSS2 I'm using the Hera paths in this description of what was used.

I have copied my WCOSS2 test logs for the gldas job to Hera for your viewing:

/scratch1/NCEPDEV/global/Kate.Friedman/stuff/GLDAS/202209*/gdasgldas.log

The most recent log (2022092000/gdasgldas.log) is for the first gldas job where I tried the new set and which failed with the error above. All other logs are from using the existing set and did not fail.

Could there be an issue with the new fracgrid T1534 fix files? Or is this because this run was warm-started (although I'm not sure if that matters this far into the run now?)? The issue documented in this issue has always been related to cold-starting a run so I didn't expect it to fail in a warm-started run that was already running gldas with the existing fix set without issues. Please take a look at the log and let me know if I should adjust any settings when Cactus is back tomorrow and I can rerun the failed job. Thanks!

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA @HelinWei-NOAA The new fix files should only be used for cold starts.  Warm starts should continue to use the old fix files.  The reason for this is that warm starts continue to use the same version of the GFS, which uses an older algorithm for the land mask and thus does not change the land mask to match what is in the new fix files.  My suggestion was to create two sets of fix files -- one for cold starts and one for warm starts.  This will require changes to the gldas scripts to point to the correct fix dataset.

[See my more detailed explanation above](https://github.com/NOAA-EMC/global-workflow/issues/622#issuecomment-1039334745).

**Comment by KateFriedman-NOAA** [comment]
@DavidHuber-NOAA Ok I see. So I should use the following in these situations then (please check my understanding):

* cold-start:
  * frac_grid on: /scratch1/NCEPDEV/global/glopara/fix/gldas/20220920/frac_grid/FIX_T1534
  * frac_grid off: /scratch1/NCEPDEV/global/glopara/fix/gldas/20220920/nofrac_grid/FIX_T1534
* warm-start (use older/existing set):
  * frac_grid on (?): /scratch1/NCEPDEV/global/glopara/fix/gldas/20220805/FIX_T1534
  * frac_grid off: ?

What about frac_grid off in warm-start? If warm-start use the same set regardless of frac_grid?

What bugs me about this still is that what happens in cold vs warm is carried through the tests even though they are both warm after the initial cycle. Seems to me that the cold vs warm difference should only exist in the first cycle and the same fix files should be used in either started run after that cycle. I'm not a forecast model expert though so this is just my view from the workflow end of things. :)

**Comment by DavidHuber-NOAA** [comment]
@KateFriedman-NOAA I believe the current fix files should work for both nofrac_grid and frac_grid warm starts since production does not modify the land-sea mask.

**Comment by DavidHuber-NOAA** [comment]
I agree with your point that the new fix files _should_ be used after the first cycle.  Why the land sea mask does not change in future warm-start cycles does not make sense to me, either.

**Comment by KateFriedman-NOAA** [comment]
> I believe the current fix files should work for both nofrac_grid and frac_grid warm starts since production does not modify the land-sea mask.

Ok, thanks @DavidHuber-NOAA for confirming. Will likely need to run nofrac_grid tests at some point.

> I agree with your point that the new fix files should be used after the first cycle. Why the land sea mask does not change in future warm-start cycles does not make sense to me, either.

@HelinWei-NOAA @junwang-noaa @yangfanglin Is there a particular reason that a different land sea mask file is used depending on how the run was started? If a cycle is using warm-starts (after the initial cold-start cycle) why is it different than a warm-started run that is also using warm-starts in it's cycles? Is it possible to only use the special cold-start GLDAS fix file set in the first gldas job of a cold-started run and then use the existing fix set for later cycles (and in all warm-started run cycles)? Hope my questions make sense. Thanks!

**Comment by KateFriedman-NOAA** [comment]
Apologies @JunDu-NOAA I meant to tag @junwang-noaa in the above comment and picked your GitHub username by accident from the list. You can ignore the ping. :)

**Comment by JunDu-NOAA** [comment]
No problem, thanks!

**Comment by yangfanglin** [comment]
@KateFriedman-NOAA @DavidHuber-NOAA  The same set of fix files should be used for all cases discussed here, no matter it is warm or cold start, first cycle or the cycles after, fractional or non-fractional grid.   If you are using the current operational GFS then you do need the "old" fix files.  Any UFS based applications should use the new fix files.   If they new files are not working then there are probably still issues.

**Comment by KateFriedman-NOAA** [comment]
> The same set of fix files should be used for all cases discussed here, no matter it is warm or cold start, first cycle or the cycles after, fractional or non-fractional grid. If you are using the current operational GFS then you do need the "old" fix files. Any UFS based applications should use the new fix files.

@yangfanglin Noted. Ok, trying to wrap my head around the different IC scenarios we have here then, which need to be supported in the workflow since we will have GFSv16 warm-starts for a while still. Clarification questions below.

Here are more details on my ICs and runs:

1) C192C96L127 run cold-started with ICs generated earlier this year using ops GFSv16 warm-start restarts. GLDAS job now works with updated fix files (only tested `frac_grid=.true.` scenario) after failing with old set.
2) C768C384L127 run with ops GFSv16 warm-start restarts (which were run through checksum (ncatted -a checksum,,d,, $FILE) for use by the updated UFS model). GLDAS jobs work with old fix files and fail with new ones (`frac_grid=.true.`).

Both sets of ICs were originally from GFSv16 ops but were different based on cold (run through chgres_cube) or warm. I ran with the Prototype-P8 tag of ufs-weather-model in both tests.

Question 1) All warm-started runs of UFS that use warm-start ICs from GFSv16 need to use the old GLDAS fix files, correct? Regardless of the checksum run on them?

Question 2) Any run of the UFS that uses either cold-start (regardless of source) or warm-start from a different GFSv17 run use the new GLDAS fix files? Is the correct?

Question 3) Is there an aspect of the ICs that I can check at the start of a run to help decide which set of fix files the workflow will use?

Question 4) Is there a way to process GFSv16 warm-starts for use with the new GLDAS fix files?

Thanks!

**Comment by yangfanglin** [comment]
@KateFriedman-NOAA We cannot run Prototype-P8 tag of ufs-weather-model  using GFS.v16 warm start files because of changes in a few physics schemes including land and microphysics etc.  GFS.v16 warm start ICs need to be converted to cold start ICs using CHGRES even if you are running at the C768L127 resolution .  We have been doing this for a while.  You can find a set of cold start ICs at the C768L127 resolution converted from GFS.v16 warm start ICs archived on HPSS at  /NCEPDEV/emc-global/5year/Fanglin.Yang/ICs/C768L127

**Comment by JunDu-NOAA** [comment]
FYI:  I forward this to Jun Wang.

**Comment by KateFriedman-NOAA** [comment]
> We cannot run Prototype-P8 tag of ufs-weather-model using GFS.v16 warm start files because of changes in a few physics schemes including land and microphysics etc.

Got it, thanks for explaining!

> GFS.v16 warm start ICs need to be converted to cold start ICs using CHGRES even if you are running at the C768L127 resolution .

Ok, I was wondering if that would be the case. Thanks!

> You can find a set of cold start ICs at the C768L127 resolution converted from GFS.v16 warm start ICs archived on HPSS at /NCEPDEV/emc-global/5year/Fanglin.Yang/ICs/C768L127

Good to know, thanks!

**Comment by KateFriedman-NOAA** [comment]
@BrettHoover-NOAA @DavidHuber-NOAA @HelinWei-NOAA @yangfanglin PR #1018 mostly fixed this issue, however PR #1009 will bring in changes to have the GLDAS job use the updated fix set. We will announce that the GLDAS job is working again after #1009 goes into global-workflow `develop`.

Thanks for your help on resolving this issue!

**Comment by RussTreadon-NOAA** [comment]
**Restart Tile Space Mismatch for C96 parallel**

Running a C96L127 parallel on Orion.    The 2021122500 gdasgldas job is the first job in the parallel with sufficient `sfluxgrb` files to run gldas.   Executable `./LIS` fails with the following error:

```
 0:  noah_scatter()
 0:  NOAH Restart File Used: noah.rst
 0:            1         384         192       24793
 0:  Restart Tile Space Mismatch, Halting..
 0:  endrun is being called
 0: application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
 0: In: PMI_Abort(1, application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0)
 0: slurmstepd: error: *** STEP 7161891.1 ON Orion-11-36 CANCELLED AT 2022-10-11T09:48:05 ***
srun: Job step aborted: Waiting up to 32 seconds for job step to finish.
```

Here are details of the parallel
- EXPDIR=/work/noaa/da/Russ.Treadon/para_gfs/prgsida4
-  HOMEgfs=/work/noaa/da/Russ.Treadon/git/global_workflow/develop  @ e915eb64
- ROTDIR=/work/noaa/stmp/rtreadon/comrot/prgsida4
- jog log file:  /work/noaa/stmp/rtreadon/comrot/prgsida4/logs/2021122500/gdasgldas.log
- job run directory:  /work/noaa/stmp/rtreadon/RUNDIRS/prgsida4/2021122500/gdas/gldas.63881

Tagging @KateFriedman-NOAA , @WalterKolczynski-NOAA , @HelinWei-NOAA 

Have we successfully run gldas on Orion for C96 after #1009 was merged into `develop`?

**Comment by DavidHuber-NOAA** [comment]
@HelinWei-NOAA @KateFriedman-NOAA @WalterKolczynski-NOAA Looking through the history, it seems we never generated the `sfc.gaussian.nemsio` file for C96 and the associated fix file for T192.  Helin, it looks like the file you would need is /work/noaa/stmp/rtreadon/RUNDIRS/prgsida4/2021122500/gdas/gldas.63881/sfc.gaussian.nemsio.20211222.

**Comment by HelinWei-NOAA** [comment]
@DavidHuber-NOAA You are right. We haven't updated the gldas fixed fields for the resolution C96. Now we have one for fraction grid (/work/noaa/stmp/rtreadon/RUNDIRS/prgsida4/2021122500/gdas/gldas.63881/sfc.gaussian.nemsio.20211222). Would you mind running another C96 case but turning off fractional grid? Thanks.

**Comment by DavidHuber-NOAA** [comment]
@HelinWei-NOAA Will do.

**Comment by DavidHuber-NOAA** [comment]
@HelinWei-NOAA OK, the C96, non-fractional grid sfc file is here: /work/noaa/nesdis-rdo2/dhuber/for_helin/96_nofrac/sfc.gaussian.nemsio.20220401.

**Comment by HelinWei-NOAA** [comment]
@DavidHuber-NOAA This nemsio file is for C192 resolution even you ran the model @C96. My program is not robust enough to fail with the wrong resolution input file. I check the previous nemsio tarball you created. The data is always one level higher resolution than it should. @KateFriedman-NOAA I need to recreate some GLDAS fixed fields because of this mismatch.

**Comment by KateFriedman-NOAA** [comment]
> I need to recreate some GLDAS fixed fields because of this mismatch.

@HelinWei-NOAA Okie dokie. Pass me the updated/new files when ready and I'll copy them into the fix set. Thanks!

**Comment by HelinWei-NOAA** [comment]
@DavidHuber-NOAA When you ran gdas@c96, gfs was always run @ 1 level higher resolution (c192 for this case). nemsio file was created by gfs?

**Comment by DavidHuber-NOAA** [comment]
@HelinWei-NOAA The resolution is correct for C96.  The GLDAS runs at T190 for a C96 case and that is the resolution of the nemsio file I provided.  The GLDAS resolution is determined by the global workflow script [scripts/exgdas_atmos_gldas.sh](https://github.com/NOAA-EMC/global-workflow/blob/develop/scripts/exgdas_atmos_gldas.sh) at line 58:
`JCAP=$((2*res-2))`

This is confirmed on lines 90 through 93 where the linking to the correct fix files is written as
```
if [[ "${FRAC_GRID:-".true."}" = ".true." ]] ; then
  ln -fs "${FIXgldas}/frac_grid/FIX_T${JCAP}" "${RUNDIR}/FIX"
else
  ln -fs "${FIXgldas}/nonfrac_grid/FIX_T${JCAP}" "${RUNDIR}/FIX"
fi
```

So I believe you have correctly derived the fix files for C192 - C768 and that between Russ' and my nemsio files, you will have what you need for C96.

If T190 is supposed to correspond to C192, then the gldas scripts need to be rewritten.

**Comment by HelinWei-NOAA** [comment]
@DavidHuber-NOAA You are absolutely right. Thank you for clarification. However, my program interpreted them wrongly, so I need to recreate the whole set of gldas fixed filed data. Do you have the nemsio files for C192 and C384 without fractional grid? Thanks.

**Comment by HelinWei-NOAA** [comment]
@DavidHuber-NOAA I found those two nemsio files. Thanks.

**Comment by HelinWei-NOAA** [comment]
@KateFriedman-NOAA @DavidHuber-NOAA It turns out only C96 (T190) has a problem. Please copy the updated data for this resolution FIX_T190 from /scratch1/NCEPDEV/global/Helin.Wei/save/fix_gldas for both nonfrac_grid and frac_grid. Thanks.

**Comment by KateFriedman-NOAA** [comment]
@HelinWei-NOAA @DavidHuber-NOAA @RussTreadon-NOAA I have rsync'd the updated FIX_190 nonfrac_grid and frac_grid files into the fix/gldas/20220920 subfolder on all supported platforms. Please retest the GLDAS job that failed and let me know if further updates are needed. Thanks!
